### PR TITLE
Add function `forNetwork()` to class `Client` (PR)

### PR DESCRIPTION
**Pull request by deyanzz**
**Description**:
This PR add the ability to construct a `Client` object, which is pre-configured for a specific local node. 
The Unit test to test this functionality is loading the configuration settings from `local_node.json` JSON file using [nlohmann JSON](https://github.com/nlohmann/json) - see test `ConnectToLocalNode` in `ClientIntegrationTest`.

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-sdk-cpp/issues/201

**Notes for reviewer**:

Suggestions for improvement:

- API doc comments could be written much better
- Test `ConnectToLocalNode` in suite `ClientIntegrationTest` should check only if a connection to local node has been established.
- An example how to connect to a local node is missing. Maybe it's a good idea to add one.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/main/include/impl/Network.h [hunk 1]
```diff
@@ -97,6 +105,15 @@ class Network
...
   void setNetwork(const NodeAddressBook& nodeAddressBook);
 
+  /**
+   * Establish communications with all node addresses for this Network that are specified in the input map collection.
+   *
+   * @param networkMap The map containing string node addresses pointing to specific accountId object.
+   * @throws UninitializedException If a NodeAddress in the input NodeAddressBook wasn't initialized with a certificate
+   *         hash.
+   */
+  void setNetwork(const std::map<std::string, Hedera::AccountId>& networkMap);
```
Comment (rwalworth) [inline sdk/main/include/impl/Network.h:115]:
```suggestion
  void setNetwork(const std::unordered_map<std::string, AccountId>& networkMap);
```

**Review by rwalworth — CHANGES_REQUESTED** [review]

File: sdk/main/include/impl/Network.h [hunk 2]
```diff
@@ -60,6 +61,13 @@ class Network
    */
   [[nodiscard]] static Network forPreviewnet();
 
+  /**
+   * Construct a Network that is pre-configured for local node access.
+   *
+   * @return A Network object that is set-up to communicate with a specific node addresses.
+   */
+  [[nodiscard]] static Network forNetwork(const std::map<std::string, Hedera::AccountId>& networkMap);
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/include/impl/Network.h]: Fixed in 53db0190768de9838818350abd213deba7203ecf.

File: sdk/main/include/impl/NodeAddressBook.h [hunk 3]
```diff
@@ -66,6 +66,15 @@ class NodeAddressBook
...
   [[nodiscard]] static NodeAddressBook fromProtobuf(const proto::NodeAddressBook& protoAddressBook);
 
+  /**
+   * Construct a NodeAddressBook object from an address map.
+   *
+   * @param addressMap The address map containing accountId to node address relations.
+   * @return The constructed NodeAddressBook object.
+   */
+  [[nodiscard]] static NodeAddressBook fromAddressMap(
+    const std::unordered_map<AccountId, std::shared_ptr<internal::NodeAddress>>& addressMap);
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/include/impl/NodeAddressBook.h:76]: Fixed in cbfba6e48466d33c4f2a3bbfbf92132dc3a6adf8.

[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/include/impl/NodeAddressBook.h]: Fixed in cbfba6e48466d33c4f2a3bbfbf92132dc3a6adf8.

File: sdk/main/src/impl/Network.cc [hunk 4]
```diff
@@ -87,6 +95,28 @@ void Network::setTLSBehavior(TLSBehavior desiredBehavior) const
...
+  std::map<std::string, Hedera::AccountId>::const_iterator it;
+
+  for (it = networkMap.begin(); it != networkMap.end(); it++)
+  {
+    const std::string& nodeAddressAsString = it->first;
+    const AccountId& accountId = it->second;
+
+    NodeAddress nodeAddress = NodeAddress::fromString(nodeAddressAsString);
+    std::shared_ptr<internal::NodeAddress> nodeAddressPtr = std::make_shared<NodeAddress>(nodeAddress);
+    nodeAddressPtr->setNodeCertHash("Test Node Certificate Hash");
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/src/impl/Network.cc]: Fixed in 4a6c6099b93429865f647b4bfe5cb79436644554.

File: sdk/main/src/impl/NodeAddressBook.cc [hunk 5]
```diff
@@ -60,6 +60,15 @@ NodeAddressBook NodeAddressBook::fromProtobuf(const proto::NodeAddressBook& prot
   return outputAddressBook;
 }
 
+//-----
+NodeAddressBook NodeAddressBook::fromAddressMap(
+  const std::unordered_map<AccountId, std::shared_ptr<internal::NodeAddress>>& addressMap)
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/src/impl/NodeAddressBook.cc]: Fixed in cbfba6e48466d33c4f2a3bbfbf92132dc3a6adf8.

File: sdk/tests/ClientTest.cc [hunk 6]
```diff
@@ -104,3 +116,48 @@ TEST_F(ClientTest, SetDefaultMaxTransactionFee)
...
+  EXPECT_NO_THROW(jsonData = json::parse(testInputFile));
+
+  if (jsonData[jsonNetworkTag][testAccountIdStr].is_string())
+  {
+    nodeAddress = jsonData[jsonNetworkTag][testAccountIdStr];
+  }
+
+  testInputFile.close();
+
+  std::map<std::string, Hedera::AccountId> networkMap;
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/tests/ClientTest.cc]: Fixed in f633752331714c769402fd69cecaee2a47a67fcf.

File: sdk/tests/ClientTest.cc [hunk 7]
```diff
@@ -104,3 +116,48 @@ TEST_F(ClientTest, SetDefaultMaxTransactionFee)
...
+  networkMap.insert(std::pair<std::string, Hedera::AccountId>(nodeAddress, AccountId::fromString(testAccountIdStr)));
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/tests/ClientTest.cc]: Fixed in f633752331714c769402fd69cecaee2a47a67fcf.

File: sdk/main/src/impl/Network.cc [hunk 8]
```diff
@@ -87,6 +95,28 @@ void Network::setTLSBehavior(TLSBehavior desiredBehavior) const
   }
 }
 
+//-----
+void Network::setNetwork(const std::map<std::string, Hedera::AccountId>& networkMap)
+{
+  std::unordered_map<AccountId, std::shared_ptr<internal::NodeAddress>> addressMap;
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/src/impl/Network.cc]: Fixed in 4a6c6099b93429865f647b4bfe5cb79436644554.

File: sdk/tests/ClientTest.cc [hunk 9]
```diff
@@ -104,3 +116,48 @@ TEST_F(ClientTest, SetDefaultMaxTransactionFee)
   // Negative value should throw
   EXPECT_THROW(client.setMaxTransactionFee(fee.negated()), std::invalid_argument);
 }
+
+//-----
+TEST_F(ClientTest, ConnectToLocalNode)
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/tests/ClientTest.cc]: Done. All of the tests have been separated in 2 different folders:

- integration
- unit

Check 0f5f4b765280fae74e1c90f6ce6a26a9570a6400 and 3e7deebe56c7781d228d571ef873f8c479249898.

File: sdk/tests/CMakeLists.txt [hunk 10]
```diff
@@ -1,80 +1,2 @@
...
-    FetchContent_Declare(googletest
-            GIT_REPOSITORY https://github.com/google/googletest.git
-            GIT_TAG release-1.12.1)
-
-    set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
-    FetchContent_MakeAvailable(googletest)
-    include(googletest)
-else ()
-    find_package(GTest REQUIRED)
-endif ()
```
Comment (rwalworth) [inline sdk/tests/CMakeLists.txt:12]: This `FetchContent` could probably just stay in this file and not be repeated in the `unit/` and `integration/` subfolder CMakeLists.txt files

**Review by rwalworth — APPROVED** [review]
LGTM, just one more fix

File: sdk/tests/CMakeLists.txt (hunk 10 above)
Comment (deyanzz) [reply sdk/tests/CMakeLists.txt:12]: Fixed in 517301850c956d3202e3d3446eab1d1a47be2e81.

File: sdk/main/src/impl/Network.cc [hunk 11]
```diff
@@ -87,6 +95,28 @@ void Network::setTLSBehavior(TLSBehavior desiredBehavior) const
...
+  std::map<std::string, Hedera::AccountId>::const_iterator it;
+
+  for (it = networkMap.begin(); it != networkMap.end(); it++)
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/src/impl/Network.cc]: Thanks for noticing it. It's fixed in 8fce51c061e78410835fa2943ccf6ab4d24b11fa.

**Review by rwalworth — APPROVED** [review]
LGTM

**Pull request merged**